File: MP.Core/Models/Purchase.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace MP.Core.Models
{
    public class Purchase
    {
        public int Id { get; set; }
        public decimal TotalNoDiscount { get; set; }
        public decimal TotalWithDiscount { get; set; }
        public decimal Discount { get; set; }
        public DateTime PurchaseDate { get; set; }
        public int NumberOfTickets { get; set; }
        public int UserId { get; set; }
        public User User { get; set; }
        public List<Ticket> Tickets { get; set; }
    }
}

File: MP.Web/ClientApp/src/models/purchase.ts
enum PurchaseFields {
    TOTAL_NO_DISCOUNT = "totalNoDiscount",
    TOTAL_WITH_DISCOUNT = "totalWithDiscount",
    DISCOUNT = "discount",
    PURCHASE_DATE = "purchaseDate",
    NUMBER_OF_TICKETS = "numberOfTickets",
    USER = "user",
    TICKETS = "tickets"
}

export default interface IPurchase {
    id?: number,
    [PurchaseFields.TOTAL_NO_DISCOUNT]: number,
    [PurchaseFields.TOTAL_WITH_DISCOUNT]: number,
    [PurchaseFields.DISCOUNT]: number,
    [PurchaseFields.PURCHASE_DATE]: Date,
    [PurchaseFields.NUMBER_OF_TICKETS]: number
}